lib/exception_handling/exception_info.rb

```diff
@@ -59,6 +59,7 @@ def initialize(exception, exception_context, timestamp, controller: nil, data_ca
       @data_callback = data_callback
       # merge into the surrounding context just like ContextualLogger does when logging
       @merged_log_context = ExceptionHandling.logger.current_context_for_thread.deep_merge(log_context || {})
+      @honeybadger_tags = [@merged_log_context[:honeybadger_tags]].flatten.compact
```

Comment:
Per my comment on the spec, the `compact` call was confusing me here. I thought you wanted to remove nils, but there wasn't a spec for that. Now I realize it's handling the special case where the key isn't given.

How about this approach instead to handle that case?
```ruby
@honeybadger_tags = [@merged_log_context[:honeybadger_tags] || []].flatten
```

Reply:
@gabe-kent Also, I just remembered that `Array()` is actually a bit cleaner than `[].flatten`. It's a bit more efficient, too, because when its argument is already an `Array`, it just passes it through with effort.
```ruby
@honeybadger_tags = Array(@merged_log_context[:honeybadger_tags] || [])
```